lib/dpla/conditions.rb

```diff
@@ -11,13 +11,19 @@ def to_s
     private
 
     def transform(conditions)
+      # When is `conditions' not an array or hash?  It seems that it should
```

Comment:
I, too like the idea of an exhaustive case statement with good default behavior. The default case doesn't, at the moment, assert that the `conditions` getting passed to `encode_uri` is a String or has `#to_string`.  I'd prefer to see it raise an exception here, or assert that this is a string; but I think this is out of scope here, and I think the most I should do is modify my comment.